ProjectEureka/project-eureka-flutter
# PE-90 | Allow Multiple Chats For A Question

This PR was for when I noticed that a user who posts multiple questions has chats active with those who tried to answer the question cannot see the multiple chats created from the question. When a user has multiple questions, all questions with its respective chats should show up. 

What I Did:

- Rewrote the group chat id creation to include the question id, leading to multiple chats available of the same question. 
- Allowed multiple chats created by a question to be visible to the user who wrote the question. 

How To test:

- Checkout to my branch PE-90, and pull the changes
- On the Home Page, start a chat with these two questions by clicking on more details, then clicking the answer button.
![Screen Shot 2021-04-28 at 2 52 18 PM](https://user-images.githubusercontent.com/48227740/116477351-55b94080-a831-11eb-956b-84b80b278a1e.png)

- Send a message
![Screen Shot 2021-04-28 at 2 54 01 PM](https://user-images.githubusercontent.com/48227740/116477504-931dce00-a831-11eb-84ac-26b0cd29f328.png)
- Once you send the message, preform the same with the other question. 
- Navigate to the home page. You should see two chats which were started from the more details page. Verify that you see the name "Hugh" as the sender and the question titles relate to the questions you answered in the previous step. 
- ![Screen Shot 2021-04-28 at 2 55 46 PM](https://user-images.githubusercontent.com/48227740/116477693-d11af200-a831-11eb-8b0e-62cb174be377.png)
- Clicking inside each bubble should show you the messages you sent from the more details page. 
- In each conversation, verify that the name "Hugh" is the one at the top of the screen. 

![Screen Shot 2021-04-28 at 2 57 06 PM](https://user-images.githubusercontent.com/48227740/116477830-01629080-a832-11eb-87b9-43e7ebab093f.png)

- Once all is done, verify with me that you have done the previous steps. If I can see your chats from my account, then the pr is successful.
